core/domain/stats_jobs_one_off_test.py

```diff
@@ -28,11 +34,663 @@
+    exp_id = 'exp_id'
+    exp_version = 1
+    state = 'state_1'
+    session_id_1 = 'session_id_1'
+    session_id_2 = 'session_id_2'
+
+    def setUp(self):
+        super(RecomputeAnswerSubmittedStatisticsTest, self).setUp()
+        stats_models.AnswerSubmittedEventLogEntryModel(
+            id='id1',
+            exp_id=self.exp_id,
+            exp_version=self.exp_version,
+            state_name=self.state,
+            session_id=self.session_id_1,
+            time_spent_in_state_secs=1.0,
+            is_feedback_useful=True,
+            event_schema_version=2).put()
+        stats_models.AnswerSubmittedEventLogEntryModel(
+            id='id2',
+            exp_id=self.exp_id,
+            exp_version=self.exp_version,
+            state_name=self.state,
+            session_id=self.session_id_1,
+            time_spent_in_state_secs=1.0,
+            is_feedback_useful=True,
+            event_schema_version=2).put()
+        stats_models.AnswerSubmittedEventLogEntryModel(
+            id='id3',
+            exp_id=self.exp_id,
+            exp_version=self.exp_version,
+            state_name=self.state,
+            session_id=self.session_id_1,
+            time_spent_in_state_secs=1.0,
+            is_feedback_useful=False,
+            event_schema_version=2).put()
+        stats_models.AnswerSubmittedEventLogEntryModel(
+            id='id4',
+            exp_id=self.exp_id,
+            exp_version=self.exp_version,
+            state_name=self.state,
+            session_id=self.session_id_1,
+            time_spent_in_state_secs=1.0,
+            is_feedback_useful=True,
+            event_schema_version=1).put()
+        stats_models.AnswerSubmittedEventLogEntryModel(
+            id='id5',
+            exp_id=self.exp_id,
+            exp_version=2,
+            state_name='b',
+            session_id=self.session_id_1,
+            time_spent_in_state_secs=1.0,
+            is_feedback_useful=True,
+            event_schema_version=2).put()
+
+        state_stats_dict = {
+            'total_answers_count_v1': 3,
+            'total_answers_count_v2': 9,
+            'useful_feedback_count_v1': 3,
+            'useful_feedback_count_v2': 9,
+            'total_hit_count_v1': 3,
+            'total_hit_count_v2': 0,
+            'first_hit_count_v1': 3,
+            'first_hit_count_v2': 0,
+            'num_times_solution_viewed_v2': 0,
+            'num_completions_v1': 3,
+            'num_completions_v2': 0
+        }
+        stats_models.ExplorationStatsModel.create(
+            self.exp_id, self.exp_version, 0, 0, 0, 0, 0, 0,
+            {
+                self.state: state_stats_dict
+            })
+        stats_models.ExplorationStatsModel.create(
+            self.exp_id, 2, 0, 0, 0, 0, 0, 0,
+            {
+                self.state: state_stats_dict
+            })
+
+    def test_standard_operation(self):
+        def mock(x, y, z): #pylint: disable=unused-argument,invalid-name
```

Comment:
Can we call this mock_get_snapshots_metadata? Ditto below.

Reply:
This ended up not being used with how the tests were updated. Done.